sphinx_gallery/tests/test_backreferences.py

```diff
@@ -89,4 +78,30 @@ def test_identify_names():
              'module_short': 'sphinx_gallery.back_references'}
     }
 
+    res = sg.identify_names('sphinx_gallery/tests/unicode.sample')
+    assert expected == res
+
+
+def test_identify_names2():
+    code_str = b"""
+# -*- coding: utf-8 -*-
+# \xc3\x9f
+from a.b import c
+import d as e
+print(c)
+e.HelloWorld().f.g
+"""
+    expected = {'c': {'name': 'c', 'module': 'a.b', 'module_short': 'a.b'},
+                'e.HelloWorld': {'name': 'HelloWorld', 'module': 'd', 'module_short': 'd'}}
+
+    import locale
+    print(locale.getlocale())
+
+    with tempfile.NamedTemporaryFile('wb', delete=False) as f:
```

Comment:
Using pytest `tmpdir` fixture is a lot more neat for this kind of things.

Reply:
I just did this of using the fixture, but all tests jobs in travis (except 1) failed

Reply:
Any info on this. It's recent commit https://github.com/sphinx-gallery/sphinx-gallery/pull/311/commits/f2f2d04e4fd7401b1ec1ffe6833716505c65c166